libr/core/cmd_anal.c

```diff
@@ -5073,8 +5073,13 @@ static void cmd_anal_syscall(RCore *core, const char *input) {
 		} else {
 			list = r_syscall_list (core->anal->syscall);
 			r_list_foreach (list, iter, si) {
-				r_cons_printf ("%s = 0x%02x.%u\n",
+				// hack for macos syscalls which are hex 0x1000000 and above and dont care about swi
+				if (si->num < 0x1000000) {
```

Comment:
Couldn't you test against the value of e asm.os instead?

Reply:
I could. How can i get from core->??->asm.os

Reply:
see `r_config_get (core->config, "asm.os");`